From: medyma <aneylovemzy@gmail.com>
Date: Thu, 15 May 2026 10:00:00 +0200
Subject: [PATCH 6.6] net: phy: sfp: add XGS-PON module quirks

Add quirks for Zaram ZXOS11NPI and HSGQ NU-STICK1.0 XGS-PON SFP+
modules (plus OEM XGSPONST200x variants): ignore TX_FAULT, and
TX_FAULT + LOS where the pins are repurposed for serial communication.

Signed-off-by: medyma <aneylovemzy@gmail.com>
---
 drivers/net/phy/sfp.c | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/drivers/net/phy/sfp.c
+++ b/drivers/net/phy/sfp.c
@@ -524,3 +524,11 @@
+	/* Zaram ZXOS11NPI XGS-PON module (SFP+) */
+	SFP_QUIRK_F("Zaram", "ZXOS11NPI", sfp_fixup_ignore_tx_fault),
+	/* HSGQ NU-STICK1.0 XGS-PON module (SFP+) */
+	SFP_QUIRK_F("HSGQ", "NU-STICK1.0", sfp_fixup_halny_gsfp),
+	/* OEM XGSPONST2001 XGS-PON module (SFP+) */
+	SFP_QUIRK_F("OEM", "XGSPONST2001", sfp_fixup_halny_gsfp),
+	SFP_QUIRK_F("OEM", "XGSPONST2000", sfp_fixup_halny_gsfp),
+
 	SFP_QUIRK_F("HALNy", "HL-GSFP", sfp_fixup_halny_gsfp),
 
 	SFP_QUIRK_F("H-COM", "SPP425H-GAB4", sfp_fixup_potron),
